### PR TITLE
[FIX] Set product supplierinfo currency

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: addons/product/migrations/9.0.1.2/post-migration.py [hunk 1]
```diff
@@ -259,6 +259,21 @@ def update_product_supplierinfo(env):
...
+    # of the suppliers
+    openupgrade.logged_query(
+        env.cr,
+        """ UPDATE product_supplierinfo ps
+        SET currency_id = pp.currency_id
+        FROM product_pricelist pp
+            JOIN ir_property ip
+                ON ip.value_reference = 'product.pricelist,'||pp.id
+            JOIN res_company rc ON ip.company_id = rc.id
+            JOIN res_partner rp ON ip.res_id = 'res.partner,'||rp.id
```
Comment (pedrobaeza) [inline addons/product/migrations/9.0.1.2/post-migration.py:272]: Shouldn't you add ` AND ON rp.id = rc.partner_id`? Any way, can you comment a bit what is the status of the ir.property records you find on v8 for this?

Comment (StefanRijnhart) [reply addons/product/migrations/9.0.1.2/post-migration.py:272]: I am not interested in the partner of the company (only in its currency), but I think you mean that we need to join the supplier partners of the pricelists with the supplier partners of the supplierinfo. That happens in `rp.id = ps.name` below (as you'll know, the supplierinfo name field is a many2one on res.partner).

On a database with hundreds of suppliers and thousands of products, I have one supplier with an alternative currency. That supplier suppliers about 50 products. The following output is therefore exactly what I expect (I rewrote the query into a select):

```
SELECT ps.id, ps.name, pp.id, pp.currency_id, ip.res_id, ip.value_reference from product_supplierinfo ps
, product_pricelist pp
    JOIN ir_property ip ON ip.value_reference = 'product.pricelist,'||pp.id
    JOIN res_company rc ON ip.company_id = rc.id
    JOIN res_partner rp ON ip.res_id = 'res.partner,'||rp.id
WHERE ps.company_id = rc.id
    AND rp.id = ps.name
    AND rc.currency_id != pp.currency_id
;



  id   | name | id | currency_id |     res_id      |   value_reference   
-------+------+----+-------------+-----------------+---------------------
 11648 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11500 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7857 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7856 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7854 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7852 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7853 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11429 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11430 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7193 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7184 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7183 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7795 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7140 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7035 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7090 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7748 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7747 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7749 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7437 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7010 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6998 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7007 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7006 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7005 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7009 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7888 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6959 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7438 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7905 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  1881 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6635 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6324 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6441 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6391 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6393 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6392 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  2599 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  1982 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
   954 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  1974 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
   261 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  2522 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  2634 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  2633 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  1981 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  1938 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11742 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11741 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7855 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  7008 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  6590 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11134 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
  2632 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
 11740 |  309 |  3 |          19 | res.partner,309 | product.pricelist,3
(55 rows)
```

Comment (pedrobaeza) [reply addons/product/migrations/9.0.1.2/post-migration.py:272]: You should filter also on `ir_property` the name field for getting the supplier pricelist, not the customer one.

Then this part is still needed? https://github.com/OCA/OpenUpgrade/pull/1469/files#diff-f2bc89f2a0b2c6a44c6eae9842821138R219

Comment (StefanRijnhart) [reply addons/product/migrations/9.0.1.2/post-migration.py:272]: Thanks, added the filter on the property name. The link you referred to sets the default value to the company currency which is fine.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**